### PR TITLE
Exposed block_index to OpenEphys interfaces

### LINKED ISSUE #692: [Feature]: block index for OpenEphysBinaryRecordingInterface
### What would you like to see added to NeuroConv?

Please add support for loading multi-block Open Ephys datasets, either one block at a time or all at once.

Right now OpenEphysBinaryRecordingInterface throws an error if you try to load a multi-block Open Ephys dataset (i.e. one with more than one experiment folder under the top-level folder). Right now as far as I can tell the only workaround would be to split out each experiment into separate folders, but that's fiddly for users with lots of multi-block recordings. With a two-line change to pass the block_index through to the underlying SpikeInterface extractor it seems to load just fine. Not sure how feasible or necessary it would be to load all blocks with a single OpenEphysBinaryRecordingInterface instance, I think the single-block solution is fine for my purposes.

### Is your feature request related to a problem?

I have a multi-experiment OpenEphys recording and I want to load it with OpenEphysBinaryRecordingInterface.

### Do you have any interest in helping implement the feature?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #692

### CODE REVIEW

File: src/neuroconv/datainterfaces/ecephys/openephys/openephysbinarydatainterface.py [hunk 1]
```diff
@@ -86,15 +92,19 @@ def __init__(
         available_streams = self.get_stream_names(folder_path=folder_path)
         if len(available_streams) > 1 and stream_name is None:
             raise ValueError(
-                "More than one stream is detected! Please specify which stream you wish to load with the `stream_name` argument. "
-                "To see what streams are available, call `OpenEphysRecordingInterface.get_stream_names(folder_path=...)`."
+                "More than one stream is detected! "
+                "Please specify which stream you wish to load with the `stream_name` argument. "
+                "To see what streams are available, call "
```
Comment (h-mayorquin) [inline src/neuroconv/datainterfaces/ecephys/openephys/openephysbinarydatainterface.py:97]: I think you should print the streams for users here given that you already calculated them.

File: tests/test_on_data/test_recording_interfaces.py [hunk 2]
```diff
@@ -409,6 +413,21 @@ def check_extracted_metadata(self, metadata: dict):
         assert metadata["NWBFile"]["session_start_time"] == datetime(2020, 11, 24, 15, 46, 56)
 
 
+class TestOpenEphysBinaryRecordingInterface_version_0_5_3_block_1_stream_1(
```
Comment (h-mayorquin) [inline tests/test_on_data/test_recording_interfaces.py:416]: I would prefer to name this test by purpose `TestOpenEphysBinaryWithBlocks` and add the provenance as a comment. But this is just my preference I see the argument for adding the provenance.

**Review by h-mayorquin — COMMENTED** [review]
This looks good to me. Thanks for sorting this out so quick. I would approve it but I am afraid of auto-merge. Let's see if the user tests this today or tomorrow and if we don't we merge. I can also take a look at the data that he shared with us.

File: tests/test_on_data/test_recording_interfaces.py [hunk 3]
```diff
@@ -409,6 +413,21 @@ def check_extracted_metadata(self, metadata: dict):
...
+    RecordingExtractorInterfaceTestMixin, TestCase
+):
```
Comment (CodyCBakerPhD) [inline tests/test_on_data/test_recording_interfaces.py:418]:
```suggestion
class TestOpenEphysBinaryRecordingInterfaceWithBlocks_version_0_6_block_1_stream_1(
    RecordingExtractorInterfaceTestMixin, TestCase
):
    """From Issue #695, exposed `block_index` argument and added tests on data that include multiple blocks."""
```

@h-mayorquin

File: src/neuroconv/datainterfaces/ecephys/openephys/openephysbinarydatainterface.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/datainterfaces/ecephys/openephys/openephysbinarydatainterface.py:97]: Feel free to add this is a separate PR, I probably won't in this one (I was just breaking up a line length violation in this change)

Comment (h-mayorquin) [reply src/neuroconv/datainterfaces/ecephys/openephys/openephysbinarydatainterface.py:97]: Makes sense.

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin User OK'd the changes

**Review by h-mayorquin — APPROVED** [review]

**Pull request merged**